pkg/virt-launcher/virtwrap/network/podinterface_test.go

```diff
@@ -329,14 +329,6 @@ var _ = Describe("Pod Network", func() {
 		Expect(err).To(BeNil())
 	}
 
-	TestRunPlug := func(driver BindMechanism, ifaceName string) {
-		err := driver.discoverPodNetworkInterface(ifaceName)
-		Expect(err).ToNot(HaveOccurred())
-
-		Expect(driver.preparePodNetworkInterface()).To(Succeed())
-		Expect(driver.decorateConfig(driver.generateDomainIfaceSpec())).To(Succeed())
-	}
-
```

Comment:
Why does this test file still exist? Each test should be moved one of the to relevant new test files `bindings_test.go` or `podnic_tests.go`.

For example "phase1 should return a `CriticalNetworkError` if pod networking fails to setup" should be moved to `podnic_test.go`. The test should be rewritten so it doesn't care about the internals of `bindings.go`. It should just verify `PlugPhase1` returns a `CriticalNetworkError` in case `bindingsMock.preparePodNetworkInterfaces` returns an error. We can also add a test testing that an error in `discoverPodNetworkInterface` doesn't cause `PlugPhase1` to return a `CriticalNetworkError`.

Reply:
This would be named integration_test.go or the like or we have to mock the bindings so really test just podNIC and not podNIC + bindings, but I think that is out of the scope of this PR.

Reply:
I believe it is exactly the scope of this PR. The `podinterface.go` was splitted and removed. The same should be done for the tests.
Leaving unit tests files called `integration_tests.go` (or `podinterface_test.go` as it is called now) doesn't make any sense to me.

Reply:
Ack, I will add some binding mocks and add real unit tests here.